src/fsharp/tast.fs

```diff
@@ -3020,15 +3031,14 @@ and
       mutable binding: NonNullSlot<Val>
       /// Indicates a reference to something bound in another CCU 
       nlr: NonLocalValOrMemberRef }
-    member x.IsLocalRef = match box x.nlr with null -> true | _ -> false
-    member x.IsResolved = match box x.binding with null -> false | _ -> true
+    member x.IsLocalRef = obj.ReferenceEquals(x.nlr, null)
+    member x.IsResolved = not (obj.ReferenceEquals(x.binding, null))
     member x.PrivateTarget = x.binding
     member x.ResolvedTarget = x.binding
 
     /// Dereference the ValRef to a Val.
     member vr.Deref = 
-        match box vr.binding with 
-        | null ->
+        if obj.ReferenceEquals(vr.binding, null) then
```

Comment:
Again there's no need for this change AFAIK.  It's not a big problem either way, it's just that we should only make changes like this if they're needed